### PR TITLE
Fixed #32007 -- Fixed queryset crash with Q() annotation and aggregation.

### PR DESCRIPTION
**Pull request by felixxm**
ticket-32007

Thanks Gordon Wrigley for the report.

Regression in 8a6df55f2dd5131282084a4edfd48f63fbf8c69a.

### CODE REVIEW

**Review by carltongibson — APPROVED** [review]
OK, nice. Thanks. 👍

File: docs/releases/3.1.2.txt [hunk 1]
```diff
@@ -16,3 +16,6 @@ Bugfixes
   attribute returned incorrectly ``True`` for ``GROUP BY`` queries (e.g.
   ``.annotate().values()``) on models with ``Meta.ordering``. A model's
   ``Meta.ordering`` doesn't affect such queries (:ticket:`31990`).
+
+* Fixed a regression in Django 3.1 where a queryset crashed if it contained an
```
Comment (carltongibson) [inline docs/releases/3.1.2.txt:20]: `s/crashed/would crash/`?

**Comment by felixxm** [comment]
@carltongibson Thanks for checking :+1:

**Comment by tolomea** [comment]
👍 Speedy

**Pull request merged**